### PR TITLE
Add midi out callback

### PR DESCRIPTION
**Pull request by mckuhei**

### CODE REVIEW

File: src/lcd.cpp [hunk 1]
```diff
@@ -103,10 +103,18 @@ void LCD_ButtonEnable(lcd_t& lcd, uint8_t enable)
 
 void LCD_SetContrast(lcd_t& lcd, uint8_t contrast)
 {
-    if (contrast > 16)
-        contrast = 16;
-    else if(contrast < 1)
-        contrast = 1;
+    if (lcd.mcu.is_jv880)
```
Comment (linoshkmalayil) [inline src/lcd.cpp:106]: `lcd` has a member which is a pointer to a `mcu_t` object
So the use is 
`lcd.mcu->is_jv880`

File: src/mcu.cpp [hunk 2]
```diff
@@ -216,7 +216,14 @@ void MCU_DeviceWrite(mcu_t& mcu, uint32_t address, uint8_t data)
     case DEV_IPRD:
         break;
     case DEV_PWM1_DTR:
-        LCD_SetContrast(*mcu.lcd, 16 - (data >> 4));
+        if (*mcu.is_jv880)
```
Comment (linoshkmalayil) [inline src/mcu.cpp:219]: `*` is used when dereferencing a pointer element.
However `is_jv880` is `int` not a pointer so `*` is not needed.
Use `mcu.is_jv880`

File: src/lcd.cpp [hunk 3]
```diff
@@ -554,6 +562,7 @@ void LCD_Update(lcd_t& lcd)
                     for (int j = 0; j < 24; j++)
                     {
                         uint8_t ch = lcd.LCD_Data[i * 40 + j];
+                        LCD_FontRenderStandard(ldc, 10 + i * 50, 4 + j * 34, ' ');
```
Comment (linoshkmalayil) [inline src/lcd.cpp:565]: `lcd` typo

File: src/main_frontend.cpp [hunk 4]
```diff
@@ -888,6 +902,8 @@ ROM management options:
     FE_PrintAudioDevices();
 }
 
+#undef main
```
Comment (linoshkmalayil) [inline src/main_frontend.cpp:905]: `#undef main` is unreliable
It gave me undefined reference error to `SDL_main`
Try `#define SDL_main` which worked for me.

**Review by linoshkmalayil — CHANGES_REQUESTED** [review]

**Comment by linoshkmalayil** [comment]
Good work!
However there are some errors I have marked 
Please do check them out
Thanks!

**Comment by mckuhei** [comment]
Sorry I used github.dev to edit so there may be errors

File: src/lcd.cpp [hunk 5]
```diff
@@ -103,10 +103,18 @@ void LCD_ButtonEnable(lcd_t& lcd, uint8_t enable)
 
 void LCD_SetContrast(lcd_t& lcd, uint8_t contrast)
 {
-    if (contrast > 16)
-        contrast = 16;
-    else if(contrast < 1)
-        contrast = 1;
+    if (lcd->mcu.is_jv880)
```
Comment (linoshkmalayil) [inline src/lcd.cpp:106]: Misplaced `->`
Should be `lcd.mcu->is_jv880`

**Review by linoshkmalayil — CHANGES_REQUESTED** [review]

**Review by linoshkmalayil — APPROVED** [review]

**Comment by linoshkmalayil** [comment]
Changes look good to me
The emulator compiles and runs as expected.

**Pull request merged**